Plutarch/List.hs

```diff
@@ -363,3 +370,134 @@ plistEquals =
         )
         (pelimList (\_ _ -> pconstant False) (pconstant True) ylist)
         xlist
+
+preverse :: (PIsListLike l a) => Term s (l a :--> l a)
+preverse = phoistAcyclic $
+  plam $ \xs -> pfoldl # plam (\ys y -> pcons # y # ys) # pnil # xs
+
+pelemAt :: (PIsListLike l a) => Term s (PInteger :--> l a :--> a)
+pelemAt = phoistAcyclic $
+  plam $ \n xs ->
+    pif
+      (n #< 0)
+      perror
+      (pelemAt' # n # xs)
+
+pelemAt' :: (PIsListLike l a) => Term s (PInteger :--> l a :--> a)
+pelemAt' = phoistAcyclic $
+  pfix #$ plam $ \self n xs ->
+    pif
+      (n #== 0)
+      (phead # xs)
+      (self # (n - 1) #$ ptail # xs)
+
+pfind :: (PIsListLike l a) => Term s ((a :--> PBool) :--> l a :--> PMaybe a)
+pfind = phoistAcyclic $
+  pfix #$ plam $ \self f xs ->
+    pelimList
+      ( \y ys ->
+          pif
+            (f # y)
+            (pcon $ PJust y)
+            (self # f # ys)
+      )
+      (pcon PNothing)
+      xs
+
+psort :: POrd a => Term s (PList a :--> PList a)
```

Comment:
When do you actually use sorting?

Reply:
I originally implemented this because I thought we would need sorting to convert values into sorted list so that functions like unionWith can be more efficient. Hopefully the values come sorted or can be required to be sorted, but we haven't tested that yet.